### PR TITLE
Added new function to issue a alert if the service is up again

### PR DESCRIPTION
**Pull request by SeaLife**
Hey there,

i've added a new function to run a alert if the service is up again.

```check_interval: 30

monitors:
  - name: My Website
    command: [ 'curl', 'https://minitor.mon' ]
    alerts: [ log, email, sms ]
    notify_up: [ sms_up, email_up ]
    check_interval: 30
    alert_after: 3
    alert_every: -1 # Defaults to -1 for exponential backoff

alerts:
  email:
    command: [ sendmail, "me@minitor.mon",  "Failure: {monitor_name}",  "This thing failed!" ]
  email_up:
    command: [ sendmail, "me@minitor.mon",  "Success: {monitor_name}",  "Service is back online! Yay!" ]
  mailgun:
    command: >
      curl -s -X POST
      -F subject="Alert! {monitor_name} failed"
      -F from="Minitor <minitor@minitor.mon>"
      -F to=me@minitor.mon
      -F text="Our monitor failed"
      https://api.mailgun.net/v3/minitor.mon/messages
      -u "api:${MAILGUN_API_KEY}"
  sms:
    command: >
      curl -s -X POST -F "Body=Failure: {monitor_name} has failed"
      -F "From=${AVAILABLE_NUMBER}" -F "To=${MY_PHONE}"
      "https://api.twilio.com/2010-04-01/Accounts/${ACCOUNT_SID}/Messages"
      -u "${ACCOUNT_SID}:${AUTH_TOKEN}"
  sms_up:
    command: >
      curl -s -X POST -F "Body=Failure: {monitor_name} is back online! :)"
      -F "From=${AVAILABLE_NUMBER}" -F "To=${MY_PHONE}"
      "https://api.twilio.com/2010-04-01/Accounts/${ACCOUNT_SID}/Messages"
      -u "${ACCOUNT_SID}:${AUTH_TOKEN}"

# federation:
#   - location: https://host1.com
#     client_key: keyfromhost1
#     server_key: keyhost1uses
#     alerts: [ sms ]

```

The sample config should clearify what i did.

Regards
SeaLife

### CODE REVIEW

**Comment by SeaLife** [comment]
Ups - failed while rebasing (squashing) my commits. Hopefully its not a problem..

**Comment by IamTheFij** [comment]
This is great! Thanks! I'll try and review/merge this week.

File: startup.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+if [ -f /app/config.sh ]; then
+  if [ ! -f /app/.config_run ]; then
+    /app/config.sh
```
Comment (IamTheFij) [inline startup.sh]: I don't see this script. What's this for?

File: README.md [hunk 2]
```diff
@@ -32,3 +32,10 @@ It will read the contents of `config.yml` and begin its loop. You could also run
 ### Configuring
 
 In this repo, you can explore the `sample-config.yml` file for an example, but the general structure is as follows. It should be noted that environment variable interpolation happens on load of the YAML file. Also, when alerts are executed, they will be passed through Python's format function with arguments for some attributes of the Monitor. Currently this is limited to `{monitor_name}`.
+
```
Comment (IamTheFij) [inline README.md]: I see. The issue with applying is that it looks like you pulled my changes and applied your changes as an amend.

You could try again by taking a new branch for your feature and then doing a `git reset --soft upstream/master`. Assuming my repo is your `upstream` remote, this should reset the references to my master, but leave all your current code in tact. Then you can re-commit it to this new branch and send a PR from that branch to my master.

Or, I could just cherry pick the parts of this that have the feature in it.

File: startup.sh (hunk 1 above)
Comment (SeaLife) [reply startup.sh]: I added this yesterday to install custom binaries into the docker container to use more command line tools to run...

You could add a config.sh in your mounted /app folder and run commands like:
```apt update
apt upgrade -y
apt install netcat```
to use nc (netcat) in your checks.. :) (The Base-Image python:2.7 does not ship with any additional command line tools)

Regards
SeaLife

**Comment by SeaLife** [comment]
Hey - i've rebased my sources onto yours and re-committed my changes so the pull-request should be fine now :)

(No history rewriting, yay!)

Regards
SeaLife

**Comment by IamTheFij** [comment]
I have a similar patch minus the Docker parts in #3.

End result should be the same, however I'm avoiding storing another bool and using a single exception that includes a reference back to the monitor that raised it. This allows determining if it is in a failing state or recovering state.

It also allows for easier access to monitor variables in the format calls. I'm going to merge later today I think.

Also looking for adding one more feature. Persisting the actual command output somewhere and then including that in the format function so you can include the check error in your notification. The output is captured today, just thrown away right now.

**Pull request closed without merging**